toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -334,3 +345,39 @@ func unbuiltPackagesCSVRows(pkgGraph *pkggraph.PkgGraph, unbuiltPackages, failed
 
 	return
 }
+
+// Helper function to print summarized numbers of the build to the logger.
+func printSummary(failedSRPMs, failedSRPMsTests map[string]*BuildResult, prebuiltSRPMs, prebuiltDeltaSRPMs, builtSRPMs, testedSRPMs, skippedSRPMsTests, unresolvedDependencies map[string]bool, blockedSRPMs, blockedSRPMsTests map[string]*pkggraph.PkgNode, rpmConflicts, srpmConflicts []string, allowToolchainRebuilds bool, conflictsLogger func(format string, args ...interface{})) {
+	logger.Log.Info("---------------------------")
+	logger.Log.Info("--------- Summary ---------")
+	logger.Log.Info("---------------------------")
+
+	logger.Log.Infof(color.GreenString("Number of prebuilt SRPMs:           " + fmt.Sprint(len(prebuiltSRPMs))))
+	logger.Log.Infof(color.BlueString("Number of prebuilt delta SRPMs:     " + fmt.Sprint(len(prebuiltDeltaSRPMs))))
+	logger.Log.Infof(color.BlueString("Number of skipped SRPMs tests:      " + fmt.Sprint(len(skippedSRPMsTests))))
+	logger.Log.Infof(color.GreenString("Number of built SRPMs:              " + fmt.Sprint(len(builtSRPMs))))
+	logger.Log.Infof(color.GreenString("Number of tested SRPMs:             " + fmt.Sprint(len(testedSRPMs))))
+	printSummaryByLen(len(unresolvedDependencies), "Number of unresolved dependencies:  ")
+	printSummaryByLen(len(blockedSRPMs), "Number of blocked SRPMs:            ")
+	printSummaryByLen(len(blockedSRPMsTests), "Number of blocked SRPMs tests:      ")
+	printSummaryByLen(len(failedSRPMs), "Number of failed SRPMs:             ")
+	printSummaryByLen(len(failedSRPMsTests), "Number of failed SRPMs tests:       ")
```

Comment:
We could simplify further by passing the slice itself and calling `len` inside the function.
```suggestion
	printSummaryByLen(unresolvedDependencies, "Number of unresolved dependencies:  ")
	printSummaryByLen(blockedSRPMs, "Number of blocked SRPMs:            ")
	printSummaryByLen(blockedSRPMsTests, "Number of blocked SRPMs tests:      ")
	printSummaryByLen(failedSRPMs, "Number of failed SRPMs:             ")
	printSummaryByLen(failedSRPMsTests, "Number of failed SRPMs tests:       ")
```

Reply:
The type of unresolvedDependencies, etc. are all different. I thought of keeping the helper function generic, so it could be extended in the future and used elsewhere also. I don't see much benefit in passing the SRPM variable, and finding length inside helper

Reply:
They are all maps and with generics the types of the keys and values shouldn't matter. Having said that, this is more a matter of preference, so no push here.:)